platform-sdk/docs/proposals/TSS-Library/native/OperatingSystem.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.pairings.nativ3support;
+
+public enum OperatingSystem {
+    WINDOWS("windows"),
+    LINUX("linux"),
+    DARWIN("darwin");
+
+    private final String directoryName;
+
+    OperatingSystem(final String directoryName) {
+        this.directoryName = directoryName;
+    }
+
+    public String directoryName() {
+        return directoryName;
+    }
+
+    /**
+     * Attempts to determine the current operating system based on the system property "os.name".
+     *
+     * @return the current operating system.
+     * @throws IllegalStateException if the current operating system is not supported.
+     */
+    public static OperatingSystem current() {
+        final String osName = System.getProperty("os.name").toLowerCase();
+        if (osName.contains("win")) {
+            return WINDOWS;
+        } else if (osName.contains("nix") || osName.contains("nux") || osName.contains("aix")) {
```

Comment:
I'm not sure AIX (or system V) would work correctly with a Linux binary.
Are we going to test those?

Reply:
This is the code we are already using:
[https://github.com/hashgraph/full-stack-testing/blob/c3fd5602525145be132770116f5bb[…]main/java/com/hedera/fullstack/base/api/os/OperatingSystem.java](https://github.com/hashgraph/full-stack-testing/blob/c3fd5602525145be132770116f5bb5a1a1922dea/fullstack-core/fullstack-base-api/src/main/java/com/hedera/fullstack/base/api/os/OperatingSystem.java)

But I'm glad to fix it in both places if it makes sense.

Reply:
We may include extras like this in test code (it's unlikely to cause a problem there), but should, in my opinion, be a bit more careful in new proposals.

Reply:
fair point